astropy/coordinates/sky_coordinate_parsers.py

```diff
@@ -643,7 +637,7 @@ def _get_representation_attrs(frame, units, kwargs):
     return valid_kwargs
 
 
-def _parse_ra_dec(coord_str):
+def _parse_ra_dec(coord_str: str, is_ra_dec_representation: bool = True) -> list[str]:
```

Comment:
Since it now does not do just ra and dec any more, I'd suggest to rename it. Maybe just `_parse_coord_str`?

Reply:
The old name really didn't make sense for the updated function anymore, so I have changed it.